recipes-wpe/wpeframework/include/compositor.inc

```diff
@@ -1,51 +1,117 @@
 
-WPE_COMPOSITOR_EXTRAFLAGS = ' \
--DPLUGIN_COMPOSITOR_AUTOSTART=${WPE_COMPOSITOR_AUTOSTART} \
--DPLUGIN_COMPOSITOR_OUTOFPROCESS=${WPE_COMPOSITOR_OUTOFPROCESS} \
--DPLUGIN_COMPOSITOR_RESOLUTION=${WPE_COMPOSITOR_RESOLUTION} \
--DPLUGIN_COMPOSITOR_MEMORY_GFX=${WPE_COMPOSITOR_MEMORY_GFX} \
--DPLUGIN_COMPOSITOR_IRMODE=${WPE_COMPOSITOR_IRMODE} \
--DPLUGIN_COMPOSITOR_VIRTUALINPUT=ON \
--DPLUGIN_COMPOSITOR_SVP="${WPE_COMPOSITOR_SVP}" \
--DPLUGIN_COMPOSITOR_BOXMODE=${WPE_COMPOSITOR_BOXMODE} \
-'
-
-WPE_COMPOSITOR_EXTRAFLAGS += "${@bb.utils.contains('WPE_COMPOSITOR_PACKAGE', 'Nexus', '-DNEXUS_SERVER_EXTERNAL=${WPE_COMPOSITOR_NEXUS_SERVER_EXTERNAL}', '', d)}"
+
+# Additional flags for Wayland implementations
 # ----------------------------------------------------------------------------
+WPE_COMPOSITOR_VIRTUALINPUT  ??= "ON"
+WPE_COMPOSITOR_CURSOR_FILE   ??= ""
 
-PACKAGECONFIG[compositor]     = " \
-    -DPLUGIN_COMPOSITOR=ON \
-    -DPLUGIN_COMPOSITOR_HARDWAREREADY=${WPE_COMPOSITOR_HW_READY} \
-    -DPLUGIN_COMPOSITOR_IMPLEMENTATION=${WPE_COMPOSITOR_IMPL} \
-    -DPLUGIN_COMPOSITOR_SUB_IMPLEMENTATION=${WPE_COMPOSITOR_SUB_IMPL} \
-    ${WPE_COMPOSITOR_EXTRAFLAGS} \
-    ,-DPLUGIN_COMPOSITOR=OFF,${WPE_COMPOSITOR_DEP} ${WPE_COMPOSITOR_PLUGIN_DEP}"
+# Choose westeros or weston dependency
+WPE_COMPOSITOR_WAYLAND_SUB_IMPL ??= "Westeros"
+WPE_COMPOSITOR_WAYLAND_DEP      ??= "westeros"
+
+# Repo details
+WPE_COMPOSITOR_WAYLAND_IMPLEMENTATION_PATH       ??= ""
+WPE_COMPOSITOR_WAYLAND_IMPLEMENTATION_REPOSITORY ??="https://code.rdkcentral.com/r/soc/broadcom/components/rdkcentral/thundernanoservices/Compositor"
+WPE_COMPOSITOR_WAYLAND_IMPLEMENTATION_VERSION    ??= "master"
 
-PACKAGECONFIG[vc4graphics] = "-DVC6=ON,,libdrm mesa"
 # ----------------------------------------------------------------------------
+PACKAGECONFIG[compositor] = "-DPLUGIN_COMPOSITOR=ON \
+    -DPLUGIN_COMPOSITOR_AUTOSTART=${WPE_COMPOSITOR_AUTOSTART} \
+    -DPLUGIN_COMPOSITOR_OUTOFPROCESS=${WPE_COMPOSITOR_OUTOFPROCESS} \
+    -DPLUGIN_COMPOSITOR_RESOLUTION=${WPE_COMPOSITOR_RESOLUTION} \
+    ,-DPLUGIN_COMPOSITOR=OFF,wpeframework-clientlibraries"
 
+PACKAGECONFIG[compositor_rpi]       = "-DPLUGIN_COMPOSITOR_IMPLEMENTATION='RPI',,"
+
+PACKAGECONFIG[compositor_wayland] = "-DPLUGIN_COMPOSITOR_IMPLEMENTATION='Wayland' \
+    -DCOMPOSITOR_WAYLAND_IMPLEMENTATION_PATH=${WPE_COMPOSITOR_WAYLAND_IMPLEMENTATION_PATH} \
+    -DCOMPOSITOR_WAYLAND_IMPLEMENTATION_REPOSITORY=${WPE_COMPOSITOR_WAYLAND_IMPLEMENTATION_REPOSITORY} \
+    -DCOMPOSITOR_WAYLAND_IMPLEMENTATION_VERSION=${WPE_COMPOSITOR_WAYLAND_IMPLEMENTATION_VERSION} \
+    -DPLUGIN_COMPOSITOR_SUB_IMPLEMENTATION=${WPE_COMPOSITOR_WAYLAND_SUB_IMPL} \
+    -DPLUGIN_COMPOSITOR_CURSOR_FILE=${WPE_COMPOSITOR_CURSOR_FILE} \
+    -DPLUGIN_COMPOSITOR_VIRTUALINPUT=${WPE_COMPOSITOR_VIRTUALINPUT},,${WPE_COMPOSITOR_WAYLAND_DEP}"
+
+PACKAGECONFIG[compositor_nexus]   = "-DPLUGIN_COMPOSITOR_IMPLEMENTATION='Nexus' \
```

Comment:
Could you please move nexus specific configuration to the meta-wpe-restricted? This includes all PACKAGECONFIGs that are nexus dependent + all flags that are used only for broadcom configurations?
IMHO the meta-wpe-restricted should have a .bbappend with those settings. This way we will have BCM specific configs in the restricted layers and for the clients nothing will change

Reply:
Yes, done